src/body_est/src/body_est/realsense.py

```diff
@@ -215,21 +171,52 @@ def pose_estimate_body(self, landmarks_list):
         # get landmarks coords
         for lm in self.lm_body:
             # transform landmark to 3D pose and transform coords to base link
-            landmarks[lm.value]["point"] = self.transform_point_baselink(
+            landmarks[lm.value]["point"] = self.transform_point_cameralink(
                 self.landmark_to_3d(landmarks[lm.value]["point"])
             )
-            # visualize points in Rviz
-            self.point_viz_pub.publish(landmarks[lm.value]["point"])
-
-        # rospy.loginfo(f"Transformed {landmarks}")
 
-    def compute_unit_normal(self, landmarks, align_vertical=False):
+        # process landmarks a bit
+        self.point_viz_pub.publish(landmarks[Landmarks.RIGHT_SHOULDER.value]["point"])
+        self.point_viz_pub.publish(landmarks[Landmarks.LEFT_SHOULDER.value]["point"])
+        self.point_viz_pub.publish(landmarks[Landmarks.RIGHT_HIP.value]["point"])
+        self.point_viz_pub.publish(landmarks[Landmarks.LEFT_HIP.value]["point"])
+
+        # returns a dictionary of the landmarks, their info, and their 3D coordinate
+        unit_nrml_x = self.compute_unit_normal(landmarks)
+        unit_nrml_y = np.array([-unit_nrml_x[1], unit_nrml_x[0], 0])
+        unit_nrml_z = np.cross(unit_nrml_x, unit_nrml_y)
+
+        rospy.loginfo(f"x {unit_nrml_x} y {unit_nrml_y} z {unit_nrml_z}")
+        rospy.loginfo(f"xy_dot {np.dot(unit_nrml_x,unit_nrml_y)}")
+        rospy.loginfo(f"xz_dot {np.dot(unit_nrml_x,unit_nrml_z)}")
+        rospy.loginfo(f"yz_dot {np.dot(unit_nrml_y,unit_nrml_z)}")
+
+        alpha = self.compute_euler_ang(unit_nrml_x, self.x_unit)
+        beta = self.compute_euler_ang(unit_nrml_y, self.y_unit)
+        gamma = self.compute_euler_ang(unit_nrml_z, self.z_unit)
+
+        torso_pt_stamped = self.est_torso_pt(landmarks)
+        self.point_viz_pub.publish(torso_pt_stamped)
+        position = torso_pt_stamped.point
+        rospy.loginfo(f"{position}")
+        q = quaternion_from_euler(alpha, beta, gamma)  # returns array of x,y,z,w
+        rospy.loginfo(f"{q}")
+        torso_pose = [position.x, position.y, position.z, q[0], q[1], q[2], q[3]]
```

Comment:
I'm trying to think of whether it is easier to return the position and orientation as a Transform. I get the sense that this code is too large to incorporate all the EKF stuff here. Personally, I would prefer if this node could be a callable service, or publish the transform as a StampedTransform, then another node can pull the data and process it.

Reply:
I think currently it is easier to publish a stamped transform with how the code is set up. However, it is also easy to be a callable service or action since that is how we implemented it for fydp. What is the more important thing here: the EKF dictating when to update or the sensor measurements choosing when to update?